darshan-util/pydarshan/tests/test_report.py

```diff
@@ -21,6 +21,32 @@ def response():
     pass
 
 
+@pytest.mark.skipif(not pytest.has_log_repo, # type: ignore
+                    reason="missing darshan_logs")
+def test_jobid_type_all_logs_repo_files(log_repo_files):
+    # test for the expected jobid type in each of the
+    # log files in the darshan_logs package;
+    # this is primarily intended as a demonstration of looping
+    # through all logs repo files in a test
+    for log_filepath in log_repo_files:
+        # darshan_logs/apmpi/darshan-apmpi-2nodes-64mpi.darshan
+        # causes an issue so we skip it here:
+        # 
+        # Error: incompatible darshan file.
+        # Error: expected version 3.21, but got 
+        # <!
+        # Error: darshan_log_open failed to read darshan log file header: Undefined error: 0.
+        if 'darshan-apmpi-2nodes-64mpi' in log_filepath:
+            continue
+        # we also need to skip apmpi data for now to avoid
+        # a segfault on pydarshan-devel with current standard
+        # config/setup options (see the CI for example)
+        if 'apmpi_apxc' in log_filepath:
+            continue
```

Comment:
@shanedsnyder the notes above re: some of the logs in the repo may be of interest--you should be able to see the steps we're using in CI to build and then reproduce the errors locally on `pydarshan-devel` branch.

My assumption here is that some of these are either newer features on `main` and/or require extra `config` stage build flags for support (feel free to suggest those). For now, I'm not too worried, but probably smoother if we can just flush through all logs unconditionally when needed for testing.

Reply:
(I'm not sure if turning on `apmpi` works super well on `pydarshan-devel` branch)

Reply:
@tylerjereddy one thing we should consider doing soon is just rebasing `pydarshan-devel` branch back onto our `main` branch? I don't think much of anything has changed related to pydarshan on `main`, so I'm hoping it wouldn't be too difficult.

I can help look into what exactly is causing the error in general, but maybe the rebase would help. We should definitely have a more graceful way of handling logs that we can't understand with specific versions of pydarshan, though. As we add logs to that repo with new format versions, it's probably important to make sure pydarshan tests don't fail on older versions. We might need to make sure we have appropriate checks for this in place, I seem to remember there being some version checking logic, but maybe it doesn't account for log format versions yet.

Reply:
@shanedsnyder Ok, I pushed in a commit to turn on autoperf submodule stuff in the Linux CI and disable the test skips here. At the moment, the DXT heatmap tests are failing for some strange reason, which is preventing me from probing this directly in CI.

@nawtrey can you help us figure out why the DXT heatmap tests are failing here? we haven't touched that code

Reply:
Yeah, I can take a look at that.

Reply:
I wonder if it is dependency versions or something--seemed "ok" when I did a quick Python 3.9 check on Linux this morning. Otherwise, something really strange

Reply:
Yeah, I think it is something like that. I did a clean install on a Linux machine and it was throwing errors on this branch as well as `pydarshan-devel`. I tried to find the dependency that could be reverted, but I haven't been able to pin it down yet. I assumed it would be `seaborn`, but updating to the latest version on my Mac install doesn't cause the tests to fail.

Reply:
It is `matplotlib`:

`python -m pip install matplotlib==3.5b1`

causes the failure for me locally, and that is the version that gets installed in a clean environment with `python setup.py install`

We really should be using `pyproject.toml` to controll installs with `python -m pip install .`, but for now we can't use that because of another issue.

In any case, probably good to investigate what has changed in the new version of `matplotlib`, and if it isn't an obvious bug, then I suppose we'll want to support the old and new approaches to whatever changed.

Reply:
For now, I'm probably going to pin `matplotlib` and open another issue to investigate this though.

Reply:
Okay, that's what I'm seeing on both of my installations, just to confirm.

Reply:
@shanedsnyder Circling back to the original problem, with the other stuff out of the way now, the CI here clearly shows the log file reading problem for a Linux matrix entry, even with the autoperf stuff turned on.

I'll try to revisit rebasing to see if that alleviates the issue now that things are slightly less chaotic with the CI/MacOS.

Reply:
So, the problem is clearly described in: https://github.com/darshan-hpc/darshan-logs/pull/5

We're dealing with an html file masquerading as a darshan binary log file in the logs repo. Once that is solved, I suspect we're actually going to be good to go here.

Reply:
this is resolved now--there is no longer an html file in place of a log file in the logs repo